src/adjoint_sensitivity.jl

```diff
@@ -261,8 +252,12 @@ function (S::AdjointSensitivityIntegrand)(out,t)
     end
     mul!(out',λ,S.pJ)
   else
-    tape = S.paramjac_config
-    vecjacobian!(out, λ, tape, y, S.p)
+    _, back = Tracker.forward(y, S.p) do u, p
+      out_ = map(zero, u)
```

Comment:
why not make this non-allocating? For small `u` this could be an issue?

Reply:
Not sure what you mean; you want `out_` to be preallocated somewhere? FWIW, this is exactly what the ReverseDiff version was doing also, the allocation of `out_` was just hidden inside the `UDerivativeWrapper`.

Reply:
I don't think it would've used UDerivativeWrapper, just UJacobianWrapper which has an in-place form of the function? But I guess the issue is that ReverseDiff couldn't use the non-allocating form? https://github.com/JuliaDiffEq/DiffEqSensitivity.jl/blob/master/src/derivative_wrappers.jl#L88 @YingboMa that would explain why it was bad on smaller models.

Reply:
Sure, whichever one – https://github.com/JuliaDiffEq/DiffEqDiffTools.jl/blob/2dc3c201ad0b1c568ce204f409ec2806000b32e4/src/function_wrappers.jl#L16

I expect it's redundant given that Flux has to allocate anyway. But we can benchmark at some point.

Reply:
Oh, I thought ReverseDiff would try to use the non-allocating form https://github.com/JuliaDiffEq/DiffEqDiffTools.jl/blob/2dc3c201ad0b1c568ce204f409ec2806000b32e4/src/function_wrappers.jl#L15 ...